pkg/elfparser/elf.go

```diff
@@ -201,12 +201,20 @@ func (e *elfLoader) loadMap(parsedMapData []ebpf_maps.CreateEBPFMapInput) (map[s
 		}
 
 		//Fill ID
-		mapInfo, err := (e.bpfMapApi).GetMapFromPinPath(pinPath)
-		if err != nil {
-			return nil, fmt.Errorf("map '%s' doesn't exist", mapNameStr)
+		if loadedMaps.PinOptions.Type == constdef.PIN_NONE.Index() {
+			mapInfo, err := (e.bpfMapApi).GetBPFmapInfo(bpfMap.MapFD)
```

Comment:
Ideally we can just use FD for both scenarios. But using pin path will get additional validation. It is fine for now and we can later optimize this..